src/ngAnimate/animateCss.js

```diff
@@ -0,0 +1,914 @@
+
+          $$jqLite.removeClass(node, staggerClassName);
+
+          gcsStaggerLookup.put(cacheKey, stagger);
+        }
+      }
+
+      return stagger || {};
+    }
+
+    var bod = $document[0].body;
+    var cancelLastRAFRequest;
+    var rafWaitQueue = [];
+    function waitUntilQuiet(callback) {
+      if (cancelLastRAFRequest) {
+        cancelLastRAFRequest(); //cancels the request
+      }
+      rafWaitQueue.push(callback);
+      cancelLastRAFRequest = $$rAF(function() {
+        cancelLastRAFRequest = null;
+        gcsLookup.flush();
+        gcsStaggerLookup.flush();
+
+        //the line below will force the browser to perform a repaint so
+        //that all the animated elements within the animation frame will
+        //be properly updated and drawn on screen. This is required to
+        //ensure that the the preparation animation is properly flushed so that
+        //the active state picks up from there. DO NOT REMOVE THIS LINE.
+        //DO NOT OPTIMIZE THIS LINE. THE MINIFIER WILL REMOVE IT OTHERWISE WHICH
+        //WILL RESULT IN AN UNPREDICTABLE BUG THAT IS VERY HARD TO TRACK DOWN AND
+        //WILL TAKE YEARS AWAY FROM YOUR LIFE.
+        var width = bod.offsetWidth + 1;
+        forEach(rafWaitQueue, function(cb) {
+          cb(width);
+        });
+        rafWaitQueue.length = 0;
+      });
+    }
+
+    return init;
+
+    function computeTimings(node, className, cacheKey) {
+      var timings = computeCachedCSSStyles(node, className, cacheKey, DETECT_CSS_PROPERTIES);
+      timings.maxDelay = Math.max(timings.animationDelay, timings.transitionDelay);
+      timings.maxDuration = Math.max(
+          timings.animationDuration * timings.animationIterationCount,
+          timings.transitionDuration);
+
+      return timings;
+    }
+
+    function init(element, options) {
+      var node = element[0];
+      options = $$animateOptions(element, options);
+
+      var temporaryStyles = [];
+      var classes = element.attr('class');
+      var styles = packageStyles(options);
+      var animationClosed;
+      var animationPaused;
+      var animationCompleted;
+      var runner;
+      var runnerHost;
+
+      if (options.duration === 0 || (!$sniffer.animations && !$sniffer.transitions)) {
+        close();
+        return;
+      }
+
+      var method = options.event && isArray(options.event)
+            ? options.event.join(' ')
+            : options.event;
+
+      // despite any of these methods being a valid event, we special case pure
+      // class-based animations so that we can avoid applying CSS blocking on
+      // the element to allow for normal transitions to work (this is how enter,
+      // leave and move can perform the `ng-EVENT` methods without causing an
+      // unexpected transition animation to occur).
+      var structural = method &&
+                        [' setClass ', ' addClass ',' removeClass '].indexOf(' ' + method + ' ') == -1;
```

Comment:
The spaces are a defensive way to prevent `indexOf` matching some special event value like:

```
// event = "resetClass";
['setClass', 'addClass','removeClass'].indexOf(method) == -1; //false
```